Story 2 Review

mallowmew opened the pull request.

RachmannJoubert committed:
Merge pull request #1 from Mayden-Academy/s2t5-design-db

reference folder created, contains database design

lizsharp committed:
reviewed catapi documentation. selected relevant information and provided in the readme file

AlisonYoon committed:
Update README.md

AlisonYoon committed:
Update README.md

AlisonYoon committed:
Merge pull request #2 from Mayden-Academy/s2t6-api-doc

s2t6-api-doc

Tyrrhene committed:
Code to Drop table if it exists, and create table in try-catchblock.

AlisonYoon committed:
fill-db file added

AlisonYoon committed:
gitignore file added

AlisonYoon committed:
variables created

Tyrrhene committed:
Now creates specified database architecture

lizsharp committed:
Added all directories and files as described in app architecture

Tyrrhene committed:
Removed two extra lines

AlisonYoon committed:
Merge pull request #6 from Mayden-Academy/s2t7-creat-dir

Added all directories and files as described in app architecture

AlisonYoon committed:
Merge branch 'story2' of github.com:Mayden-Academy/2019-top-cat into s2t2-fill-db

AlisonYoon committed:
parking to pull Liz's change

AlisonYoon committed:
DB connection method created inside DB class

lizsharp committed:
Removed api content fro 2019-top-cat README.md and put it in APIScrape/README.md

Tibereon committed:
Merge pull request #5 from Mayden-Academy/s1t2-bd-build

S1t2 bd build

AlisonYoon committed:
getCatBreed method returns an array of cat breed

AlisonYoon committed:
Merge branch 'story2' of github.com:Mayden-Academy/2019-top-cat into s2t2-fill-db

AlisonYoon committed:
getCatImg method now get the array of catImgUrls. Will use this to get response of img src

lizsharp committed:
Added request parameters and expected key response info. Added pagination and ordering info.

MotNosniktaw committed:
Merge pull request #7 from Mayden-Academy/s2t6-relocate-api-readme

s2t6-relocate-api-readme

AlisonYoon committed:
Get img url from imgAPI

AlisonYoon committed:
Merge branch 'story2' of github.com:Mayden-Academy/2019-top-cat into s2t2-fill-db

lizsharp committed:
Created gitignore file

lizsharp committed:
Added vendor to gitignore

lizsharp committed:
Added composer.json file

AlisonYoon committed:
getCatImg method gets an array of all the cat image url of all the breeds the CatAPI provides

lizsharp committed:
Added composer.lock and vendor package

AlisonYoon committed:
filled DB breed table

FelixLetheren committed:
Merge pull request #10 from Mayden-Academy/s2t7-fix-create-dir

S2t7 fix create dir

Commit:
Merge pull request #11 from Mayden-Academy/add-composer-json

Add composer json

AlisonYoon committed:
getCatBreed method returns an associative array of key id and name as value
q≈

AlisonYoon committed:
getCatImg method returns an associative array of name as key and img url as values

AlisonYoon committed:
Fill the img table with every breeds img url worksgit add Utilities/DB.phpgit add Utilities/DB.phpgit add Utilities/DB.phpgit add Utilities/DB.phpgit add Utilities/DB.phpgit add Utilities/DB.phpgit add Utilities/DB.phpgit add Utilities/DB.phpgit add Utilities/DB.phpgit add Utilities/DB.phpgit add Utilities/DB.php

AlisonYoon committed:
Deleted commented lines

AlisonYoon committed:
Doc Blocks added

AlisonYoon committed:
there was typo

AlisonYoon committed:
added space on the both side of fat arrow

AlisonYoon committed:
deleted all the unnecessary new lines

AlisonYoon committed:
made a new DB object and calling the methods. Tested, it does its job nicely

AlisonYoon committed:
Merge branch 'story2' of github.com:Mayden-Academy/2019-top-cat into s2t2-fill-db

AlisonYoon committed:
typo in variable fixed

FelixLetheren committed:
Merge pull request #13 from Mayden-Academy/s2t2-fill-db

S2t2 fill db

Commit:
Get it all in one file

Commit:
Refactor database creation

Commit:
rename getCatBreeds

Commit:
Refactor creation of assoc array

Commit:
Change DB access URL to local connection

Commit:
Remove var_dump

Commit:
Refactor fillCatBreedToDB

Commit:
Add helpful echo to fill breeds

Commit:
Add a couple newlines for readability

Commit:
Add type hinting to getCatImg

Commit:
Fix fillCatImg

Commit:
Delete db-build

Commit:
Remove extra newlines

Commit:
Fix inconsistency

AlisonYoon committed:
contents added in README file

AlisonYoon committed:
Merge branch 's2t3-cleanup' of github.com:Mayden-Academy/2019-top-cat into s2t4-readme

Commit:
Update docblocks AND helpful confirmation message

Commit:
Revert URL change in DB class

Commit:
Update docblocks

AlisonYoon committed:
scrape.php document added in README

Commit:
Tidying up

FelixLetheren committed:
Merge pull request #15 from Mayden-Academy/s2t3-cleanup

S2T3 Clean Up Scrape Script

AlisonYoon committed:
document for DB.php added

AlisonYoon committed:
Merge branch 'story2' of github.com:Mayden-Academy/2019-top-cat into s2t4-readme

AlisonYoon committed:
sql file inside ref directory document added in README

AlisonYoon committed:
Update README.md

AlisonYoon committed:
Update README.md

AlisonYoon committed:
Update README.md

AlisonYoon committed:
Update README.md

AlisonYoon committed:
Update README.md

AlisonYoon committed:
Update README.md

AlisonYoon committed:
Update README.md

AlisonYoon committed:
Update README.md

AlisonYoon committed:
Update README.md

AlisonYoon committed:
Update README.md

AlisonYoon committed:
Update README.md

AlisonYoon committed:
Update README.md

AlisonYoon committed:
made changes in README followed by feedback

AlisonYoon committed:
made changes in README followed by feedback

AlisonYoon committed:
trying to push

AlisonYoon committed:
merge conflict fixed

Commit:
Merge pull request #16 from Mayden-Academy/s2t4-readme

S2t4 readme

FelixLetheren commented on APIScrape/scrape.php:
```diff
@@ -0,0 +1,155 @@
+ */
+function createDatabase(PDO $db) {
+    try {
+        //$conn = new PDO("mysql:host=192.168.20.20", $username, $password);
```
Commented out lines

mallowmew replied:
I'll do it

FelixLetheren committed:
Updated README and removed commented lines creating DB.

AlisonYoon committed:
Merge pull request #18 from Mayden-Academy/fix-update-README

Updated README and removed commented lines creating DB.

luciavelasco commented on index.php:
```diff
@@ -0,0 +1 @@
+<?php
```
Why is this unused file in this PR?

AlisonYoon replied:
all the files that will be used in the project were created at the very beginning stage for everyone's convenience

luciavelasco replied:
Remove this file, too.

luciavelasco commented on Hydrators/CatHydrator.php:
```diff
@@ -0,0 +1 @@
+<?php
```
Why is this unused file in this PR?

AlisonYoon replied:
all the files that will be used in the project were created at the very beginning stage for everyone's convenience

luciavelasco commented on Entities/CatEntityTest.php:
```diff
@@ -0,0 +1 @@
+<?php
```
Why is this unused file in this PR?

AlisonYoon replied:
all the files that will be used in the project were created at the very beginning stage for everyone's convenience

luciavelasco replied:
Remove this file, too.

luciavelasco commented on Entities/CatEntity.php:
```diff
@@ -0,0 +1,7 @@
+<?php
```
Why is this unused file in this PR?

AlisonYoon replied:
all the files that will be used in the project were created at the very beginning stage for everyone's convenience

luciavelasco replied:
That's makes sense!
Don't make things before they're used, please remove them from this PR.

AlisonYoon replied:
ok

luciavelasco commented on APIScrape/README.md:
```diff
@@ -0,0 +1,70 @@
+TheCatApi // Developer Experience
```
# Where's the title?

This document should have some titles - it won't block the merge because it's just styling/readability, not functionality, but for future reference use:
`# Title` syntax and add more `##`s for smaller titles (in markdown).

luciavelasco commented on APIScrape/README.md:
```diff
@@ -0,0 +1,70 @@
+TheCatApi // Developer Experience
```
This readme doesn't tell me how to do the API Scrape. It tells me how to use someone else's API.
I need documentation (instructions) for using YOUR code.

AlisonYoon replied:
The README that tells you how to do the API Scrape is in the root folder.

mallowmew replied:
I'll do this

luciavelasco commented on APIScrape/scrape.php:
```diff
@@ -0,0 +1,153 @@
+<?php
+
+require_once "../Utilities/DB.php";
```
you should be using composer's autoloader in this project

luciavelasco commented on APIScrape/scrape.php:
```diff
@@ -0,0 +1,153 @@
+ */
+function createDatabase(PDO $db) {
+    try {
+        $sql = "DROP DATABASE IF EXISTS `cat-test`;
```
why is it a test?

Tibereon replied:
on It.

Tibereon replied:
cat-test has been signed off as our database name.

luciavelasco replied:
I'm actually not sure it has? It's not in the database design document

luciavelasco commented on APIScrape/scrape.php:
```diff
@@ -0,0 +1,153 @@
+        $sql = "DROP DATABASE IF EXISTS `cat-test`;
+                CREATE DATABASE `cat-test`;
+                
+                use `cat-test`;
```
Please use consistent casing: capitalise `use`.

Tibereon replied:
on it.

luciavelasco commented on APIScrape/scrape.php:
```diff
@@ -0,0 +1,153 @@
+        ),
+    ));
+    $response = curl_exec($curl);
+    $error = curl_error($curl);
```
Handle this error

AlisonYoon replied:
on it

luciavelasco commented on APIScrape/scrape.php:
```diff
@@ -0,0 +1,153 @@
+function fillCatBreedToDB(PDO $db, array $catBreeds)
+{
+    foreach($catBreeds as $id => $breed) {
+        $sql = $db->prepare('INSERT INTO `breed` (breed) VALUES (\'' . $breed . '\');');
```
You need to use parameter binding for safe PDO

AlisonYoon replied:
on it

RachmannJoubert replied:
on it

luciavelasco commented on APIScrape/scrape.php:
```diff
@@ -0,0 +1,153 @@
+
+/**
+ * Use the list of cat breeds to make the API requests so we can get URLs
+ * of cat pictures to hotlink in the app.
```
I don't think hotlink is accurate in this context. Maybe just link?

RachmannJoubert replied:
on it

Commit:
Update readme

luciavelasco commented on APIScrape/scrape.php:
```diff
@@ -0,0 +1,153 @@
+ */
+function getCatImgURLs(array $catBreeds):array
+{
+    $imgSrcArray = [];
```
Don't abbreviate variable names. Are you gaining anything from removing 5 characters?

Tibereon replied:
om it

Commit:
Fix link in docblock

Commit:
Merge branch 'fix-link-in-docblock' into story2

luciavelasco commented on APIScrape/scrape.php:
```diff
@@ -0,0 +1,153 @@
+            ),
+        ));
+        $imgApiResponse = curl_exec($curl);
+        $error = curl_error($curl);
```
handle this error

AlisonYoon replied:
on it

luciavelasco commented on APIScrape/scrape.php:
```diff
@@ -0,0 +1,153 @@
+            CURLOPT_CUSTOMREQUEST => "GET",
+            CURLOPT_RETURNTRANSFER => true,
+            CURLOPT_HTTPHEADER => array(
+                'x-api-key: ec254e44-3996-458b-8522-4933954d8fcd'
```
You're using this very complicated string in two places and it has to remain the same wherever it's used. Can we make this a constant?

mallowmew replied:
on it

luciavelasco commented on APIScrape/scrape.php:
```diff
@@ -0,0 +1,153 @@
+
+        $responseArray = json_decode($imgApiResponse, true);
+
+        $breedImgs = [];
```
Please restore the 2 missing characters to their rightful homes. It's getting cold outside!

Tibereon replied:
oN IT

Tyrrhene committed:
Consistent capitilisation, no longer test db

luciavelasco commented on APIScrape/scrape.php:
```diff
@@ -0,0 +1,153 @@
+ * 
+ * @param PDO $db the database to use
+ * @param array The list of cat breeds
+ * @param array The list of cat img URLs
```
Someone kidnapped an `a`! And an `e`! 👮‍♀ 🚔

Tibereon replied:
ON IT

luciavelasco commented on APIScrape/scrape.php:
```diff
@@ -0,0 +1,153 @@
+            $sqlArray[] = "('$url', $breedID)";
+        }
+    }
+    $sqlString = implode(',', $sqlArray);
```
NOICE

Tyrrhene committed:
Changed to signed to signed off naming architecture

Commit:
Make API key into a constant

Tibereon committed:
Merge pull request #20 from Mayden-Academy/fix-test-capitalisation-scrape.php

Consistent capitilisation, no longer test db

Commit:
Merge pull request #22 from Mayden-Academy/fix-api-key-const

Make API key into a constant

luciavelasco commented on APIScrape/scrape.php:
```diff
@@ -0,0 +1,153 @@
+$breeds = getCatBreeds();
+fillCatBreedToDB($dbconnection, $breeds);
+$catImgSrcArray = getCatImgURLs($breeds);
+fillCatImgs($dbconnection, $breeds, $catImgSrcArray);
```
All of this stuff is being run procedurally at the bottom of a functional file, and the file itself is being run manually, as a one-off, independently of your app functionality. Where's the value in making this functional? What's being reused?

luciavelasco commented on APIScrape/README.md:
```diff
@@ -0,0 +1,85 @@
+* Makes API requests to get:
+    * A list of cat breeds
+    * Image URLs for cats from each breed
+* **To run the script your mySQL server must already have a database named `cat-test`!!**
```
I don't think this matches the code below?

mallowmew replied:
It does, because of a dependency on DB.php.

When DB.php makes the PDO connection it throws up if it can't find a database named `cat-test`, because of line 5 specifying the dbname. 

DB.php could be changed but I'm not sure how this affects the app & team determined we would add this line to the readme instead.

luciavelasco replied:
Hm. We don't need a database to connect to MySQL. How about changing the DB class so we can do both/either?

mallowmew replied:
Agree but want to know the team agrees on the way forward.

mallowmew replied:
Team has decided to create another function in DB.php to connect to mySQL without specifying dbname, to be used specifically in the scrape script, so it can do its database setup stuff.

Will do this tomorrow.

AlisonYoon committed:
handle cURL errors

luciavelasco commented on README.md:
````diff
@@ -1 +1,30 @@
+* Then type in the command line 
+**(You must be inside the APIScrape directory)**
+
+    ```$ php scrape.php ```
````
Hadn't spotted this earlier! @mallowmew this looks good, but I think the edits you made to the readme in the APIScrape are better. Let's keep both to save time, but be aware that in a large codebase you'd have to be wary of one being updated without the other.

RachmannJoubert committed:
Merge pull request #23 from Mayden-Academy/fix-handle-curl-errors

handle cURL errors

Tyrrhene committed:
Removed allun uneeded abbreviations of image

Tyrrhene committed:
Removed all unnessecary abbreviations of source

FelixLetheren committed:
Merge branch 'story2' into fix-test-capitalisation-scrape.php

FelixLetheren committed:
Merge pull request #24 from Mayden-Academy/fix-test-capitalisation-scrape.php

Fix test capitalisation scrape.php

AlisonYoon committed:
unused files deleted

FelixLetheren committed:
Merge pull request #25 from Mayden-Academy/fix-delete-unused-files

unused files deleted

luciavelasco commented on APIScrape/scrape.php:
```diff
@@ -0,0 +1,161 @@
+    if($response === false) {
+        echo 'Curl error: ' . curl_error($curl);
+    } else {
+        echo 'Received cat breed successfully';
```
`breed` should be plural

Commit:
Add new function to DB to connect to host only

Commit:
Update scrape.php and improve cli reporting
scrape.php now uses a PDO without a preset database, so it can
create its own database.
Also made the CLI experience a little more polished.

Commit:
Update README

Commit:
Update the other README

Commit:
Merge pull request #26 from Mayden-Academy/fix-scrape-db-thing

Change how scraper uses DB.php

Commit:
Remove redundant USE statement

Commit:
Added parameter binding

luciavelasco commented on APIScrape/scrape.php:
```diff
@@ -58,7 +57,7 @@ function getCatBreeds():array
         echo 'Curl error: ' . curl_error($curl);
     } else {
-        echo 'Received cat breed successfully';
+        echo "Received list of cat breeds successfully.\n";
```
I like this! Standardising which quotes you use - `'` vs `"` can be a surprisingly valuable conversation for a codebase.

luciavelasco commented on APIScrape/scrape.php:
```diff
@@ -150,12 +153,12 @@ function fillCatImages(PDO $db, array $catBreedArray, array $catImageSourceArray
 
 // Do the business:
-createDatabase($dbconnection);
+createDatabase($dbConnection);
```
You fixed the casing! I didn't even notice! ⬆️5️⃣😎

luciavelasco commented on Utilities/DB.php:
```diff
@@ -12,7 +13,15 @@ class DB
     public function dbConnect()
     {
-        $dbconnect = new PDO($this->serverdb, $this->username, $this->password);
+        $dbconnect = new PDO($this->server .$this->db, $this->username, $this->password);
```
Non-blocking: missing a space after concatenation dot

RachmannJoubert replied:
on it

AlisonYoon committed:
using bindParam for sanitization

Commit:
Fixed concatination issue

AlisonYoon committed:
merge conflict fixed

luciavelasco commented on APIScrape/scrape.php:
```diff
@@ -0,0 +1,164 @@
+        if($imageApiResponse === false) {
+            echo 'Curl error: ' . curl_error($curl);
+        } else {
+            echo "\033[2KReceived cat image URLs for breed '$name' successfully ($i of $breedCount).\r";
```
https://emojis.slackmojis.com/emojis/images/1563481443/6027/meow_wow.png?1563481443
I've never seen this before, but it's f*n cool. Please add a comment to explain what it's doing, with a link to documentation or more info (perhaps bash special character documentation or a stackoverflow question would do).

AlisonYoon replied:
on it

luciavelasco commented on APIScrape/scrape.php:
```diff
@@ -0,0 +1,164 @@
+ * @param array of cat breeds
+ * @return array of breed names => array of strings representing the image URLs
+ */
+function getCatImgURLs(array $catBreeds):array
```
Your `a`s and your `e`s or your life 🔫💰🅰️

luciavelasco replied:
Cmd + Shift + R - `Img` -> `Image` (don't try that on lower case because you have `img` in your table name)

AlisonYoon replied:
on it

RachmannJoubert committed:
Merge pull request #27 from Mayden-Academy/fix-bind-params

Fix bind params

AlisonYoon committed:
Merge pull request #29 from Mayden-Academy/fix-concatination

Fixed concatination issue

AlisonYoon committed:
added a and e in the method

Commit:
Add comments describing the bash escapes

RachmannJoubert committed:
Merge pull request #30 from Mayden-Academy/fix-words-add-comments

Fix words add comments

luciavelasco commented on APIScrape/scrape.php:
```diff
@@ -114,6 +114,10 @@ function getCatImgURLs(array $catBreeds):array
         } else {
+            // Little bash technique to clean up console reporting
+            // Use bash escape \033[2K to clear the line before writing the message
+            // \r carriage returns to the beginning of the line before writing the next
```
👍

luciavelasco commented on APIScrape/scrape.php:
```diff
@@ -0,0 +1,176 @@
+ * Creates a new database into which to put the scraped data.
+ * If the database already exists, tears it down and creates a new one.
+ */
+function createDatabase(PDO $db) {
```
Missing $db param in docblock

FelixLetheren replied:
On it

luciavelasco commented on APIScrape/scrape.php:
```diff
@@ -0,0 +1,176 @@
+{
+    foreach($catBreeds as $id => $breed) {
+        $sql = $db->prepare('INSERT INTO `breed` (breed) VALUES (:breedToString);');
+        $breedToString = "$breed";
```
This is redundant, why are you not using $breed in line 85? Rename `:breedToString` to breedname

FelixLetheren replied:
On it

luciavelasco commented on APIScrape/scrape.php:
```diff
@@ -0,0 +1,176 @@
+        ));
+        $imageApiResponse = curl_exec($curl);
+        if($imageApiResponse === false) {
+            echo 'Curl error: ' . curl_error($curl);
```
This isn't handling the error, it's just showing me the error string and continuing the script without the data. Make the script exit on error.

luciavelasco replied:
don't forget to do this below, too

Tibereon replied:
on it

FelixLetheren committed:
Changed Docblock to mention $db paramater.

RachmannJoubert committed:
Merge pull request #31 from Mayden-Academy/fix-docblock

Changed Docblock to mention $db paramater.

luciavelasco commented on APIScrape/scrape.php:
```diff
@@ -0,0 +1,176 @@
+    $sqlArray = [];
+    for($breedIndex = 0; $breedIndex < count($catImageSourceArray); $breedIndex++) {
+        foreach($catImageSourceArray[$catBreedIndexedArray[$breedIndex]] as $url) {
+            $breedID = $breedIndex + 1;
```
Could you add a comment explaining that this value matches the breed id in the table?

AlisonYoon replied:
on it

luciavelasco commented on APIScrape/scrape.php:
```diff
@@ -8,7 +8,8 @@
 /**
- * Creates a new database into which to put the scraped data.
+ * @param PDO $db Takes DB containing a connection to the database.
+ *  * Creates a new database into which to put the scraped data.
```
Don't bother changing this, it doesn't block merge, but param should go after description:
```
* Creates a new database into which to put the scraped data.
*  
* @param PDO $db Takes DB containing a connection to the database.
```

FelixLetheren committed:
Removed redunded line turning $breed into another variable.

FelixLetheren committed:
Revert "Removed redunded line turning $breed into another variable."

This reverts commit 35e145126a0df364f622ec9d4333722161a35811.

Tyrrhene committed:
script exit on API error

FelixLetheren committed:
Merge branch 'story2' of github.com:Mayden-Academy/2019-top-cat into story2

luciavelasco commented on APIScrape/scrape.php:
```diff
@@ -0,0 +1,177 @@
+$breeds = getCatBreeds();
+fillCatBreedToDB($dbConnection, $breeds);
+$catImageSourceArray = getCatImageURLs($breeds);
+fillCatImages($dbConnection, $breeds, $catImageSourceArray);
```
Just putting this down for posterity:
Ultimately, I'm happy with this like this, because now we're not calling `getCatBreeds` twice. We discussed that I'd prefer a more procedural file because this is a one-off, manually run script. A lot of these functions are redundant (no potential for reuse) and the file could be shorter.
Good work! 👍

FelixLetheren committed:
Removed not needed line of turnung $breed into a string.

AlisonYoon committed:
added comment explaining why 1 is added to breedIndex

luciavelasco commented on APIScrape/scrape.php:
```diff
@@ -81,8 +81,7 @@ function fillCatBreedToDB(PDO $db, array $catBreeds)
         $sql = $db->prepare('INSERT INTO `breed` (breed) VALUES (:breedToString);');
-        $breedToString = "$breed";
-        $sql->bindParam('breedToString', $breedToString, PDO::PARAM_STR);
+        $sql->bindParam('breedToString', $breed, PDO::PARAM_STR);
```
not blocking merge, but this should just be `'breed'`, not `'breedToString`

luciavelasco replied:
Ah, reverted, nvm

luciavelasco replied:
Seems to have been implemented the same way again. `:breedToString` -> `breed` pls

FelixLetheren replied:
on it again

RachmannJoubert committed:
Merge pull request #33 from Mayden-Academy/fix-add-comment

added comment explaining why 1 is added to breedIndex

luciavelasco commented on APIScrape/scrape.php:
```diff
@@ -0,0 +1,177 @@
+ */
+function createDatabase(PDO $db) {
+    try {
+        $sql = "DROP DATABASE IF EXISTS `cat-test`;
```
This name was not signed off in db architecture, and it's not an appropriate name for the database because it's not a test. Please change it - suggest `cats`.

luciavelasco replied:
Cmd + Shift + R - `cat-test` -> `cat`

MotNosniktaw replied:
on it

FelixLetheren replied:
On it

MotNosniktaw replied:
done

AlisonYoon committed:
Merge pull request #34 from Mayden-Academy/fix-redudent-bind-param

Removed not needed line of turnung $breed into a string.

luciavelasco commented on APIScrape/scrape.php:
```diff
@@ -153,7 +153,7 @@ function fillCatImages(PDO $db, array $catBreedArray, array $catImageSourceArray
     for($breedIndex = 0; $breedIndex < count($catImageSourceArray); $breedIndex++) {
         foreach($catImageSourceArray[$catBreedIndexedArray[$breedIndex]] as $url) {
-            $breedID = $breedIndex + 1;
+            $breedID = $breedIndex + 1;     //add 1 to $breedIndex so that it matches the breed_id in DB's `breed` table
```
![yesgood](https://user-images.githubusercontent.com/14294127/65232891-d9de0a80-dac9-11e9-9743-022e59a73e3d.gif)

MotNosniktaw committed:
renamed DB to 'cats'

RachmannJoubert committed:
Merge pull request #35 from Mayden-Academy/fix-cats-db-name

renamed DB to 'cats'

AlisonYoon committed:
unused files deleted

RachmannJoubert committed:
Merge pull request #37 from Mayden-Academy/fix-remove-unused-files

unused files deleted

FelixLetheren committed:
Refactored SQL query to also just use :breed.

AlisonYoon committed:
Merge pull request #38 from Mayden-Academy/fix-redudent-bind-param

Refactored SQL query to also just use :breed.

Tyrrhene committed:
Fixed poor casing

Tibereon committed:
Merge pull request #32 from Mayden-Academy/fix-scrape-make-exit-on-error

script exit on API error

luciavelasco commented on APIScrape/scrape.php:
```diff
@@ -0,0 +1,178 @@
+    ));
+    $response = curl_exec($curl);
+    if($response === false) {
+        $curl_ExecuteError = 'Curl error: ' . curl_error($curl);
```
This is the casing you haven't fixed `$curl_ExecuteError` -> `$curlExecuteError`

Tyrrhene committed:
removed underscore

Tibereon committed:
Merge pull request #39 from Mayden-Academy/fix-scrape-make-exit-on-error

removed underscore

luciavelasco reviewed (approved):
# WE DID IT*

![b6170237eaf41bab](https://user-images.githubusercontent.com/14294127/65234560-bbc5d980-dacc-11e9-9860-970d309ca43e.gif)

###### *(you did it)

luciavelasco commented:
```bash
STORY_2 &&
echo 'Sir Isaac Newton is credited with creating the concept for the pet door that many cats use today to travel outdoors.' &&
echo 'A group of cats is called a clowder.'
```
